core/src/miner/mem_pool.rs

```diff
@@ -673,138 +726,269 @@ impl MemPool {
             .collect::<Vec<_>>();
         let fetch_seq =
             |a: &Public| signers.get(a).expect("We fetch details for all signers from both current and future").seq;
-        for hash in invalid {
-            self.remove(&hash, &fetch_seq, current_time, timestamp);
-        }
+        self.remove(&invalid, &fetch_seq, current_time, current_timestamp);
     }
 
     /// Removes invalid transaction identified by hash from pool.
     /// Assumption is that this transaction seq is not related to client seq,
     /// so transactions left in pool are processed according to client seq.
     ///
     /// If gap is introduced marks subsequent transactions as future
-    pub fn remove<F>(&mut self, tx_hash: &H256, fetch_seq: &F, current_time: PoolingInstant, timestamp: u64)
-    where
+    pub fn remove<F>(
```

Comment:
I'm happy to see the `remove` function get many transaction hashes, not just a tx_hash.